core/src/main/java/io/grpc/internal/PickFirstLeafLoadBalancer.java

```diff
@@ -609,26 +611,71 @@ public PickResult pickSubchannel(PickSubchannelArgs args) {
     }
   }
 
+  interface Index {
+    static Index create(List<EquivalentAddressGroup> groups) {
+      if (PickFirstLoadBalancerProvider.isEnabledHappyEyeballs()) {
+        return new IndexHappyEyeballs(groups);
```

Comment:
I don't see why we need two implementations. We can flatten the EAGs all the time, and for Happy Eyeballs do the extra interleaving. Only the interleaving needs to be different. I don't want to duplicate this wide API surface _just_ for interleaving.